curator-framework/src/main/java/org/apache/curator/framework/imps/EnsembleTracker.java

```diff
@@ -170,8 +170,10 @@ public static String configToConnectionString(QuorumVerifier data) throws Except
             {
                 sb.append(",");
             }
-            InetSocketAddress address = Objects.firstNonNull(server.clientAddr, server.addr);
-            sb.append(address.getAddress().getHostAddress()).append(":").append(address.getPort());
+            InetSocketAddress clientAddress = server.clientAddr;
```

Comment:
After ZooKeeper 3.5, if the zoo.cfg is in old format which has clientPort and server list being defined, then the clientAddr will be 0.0.0.0, which shouldn't be used as the server hostAddress. CURATOR-392 reported some issues similar, and it has handled this case.